library/Icingadb/Data/PivotTable.php

```diff
@@ -0,0 +1,450 @@
+<?php
+
+namespace Icinga\Module\Icingadb\Data;
+
+use Icinga\Application\Icinga;
+use Icinga\Module\Icingadb\Widget\JoystickPaginator;
+use ipl\Orm\Query;
+use ipl\Stdlib\Contract\Paginatable;
+use ipl\Stdlib\Filter;
+
+class PivotTable
```

Comment:
Still didn't have a detailed look here, but will soon. In the meantime, please make sure the code style is correct and remove any unused parts. (e.g. the order stuff)

Code style to look out for:
* array syntax
* empty lines after control structures if the body continues
* indentation